lib/base-compiler.ts

```diff
@@ -2238,11 +2235,11 @@ export class BaseCompiler implements ICompiler {
     }
 
     sanitizeCompilerOverrides(overrides: ConfiguredOverrides): ConfiguredOverrides {
-        const allowedRegex = /^[A-Z_]{1,}[A-Z0-9_]*$/;
+        const allowedRegex = /^[A-Z_]+[\dA-Z_]*$/;
         for (const override of overrides) {
             if (override.name === CompilerOverrideType.env && override.values) {
                 // lowercase names are allowed, but let's assume everyone means to use uppercase
-                override.values.forEach(env => (env.name = env.name.trim().toUpperCase()));
```

Comment:
It surprised me to read for..of is faster (and generally better) than forEach. There are ~40 instances of `forEach` in the frontend, if not via eslint - we might want to fix them manually.